extensions/common/context.cc

```diff
@@ -355,5 +433,43 @@ google::protobuf::util::Status extractNodeMetadataValue(
   return google::protobuf::util::Status(google::protobuf::util::error::OK, "");
 }
 
+bool FlatNodeCache::updatePeer(absl::string_view key, absl::string_view peer_id,
+                               absl::string_view peer_content) {
+  std::string id = std::string(peer_id);
```

Comment:
if flatmap is used, cache_ should support promoting from string_view to string

Reply:
IIRC, flatmap cannot easily run in a envoy wasm module, due to missing system call. https://github.com/istio/proxy/commit/35f3727fdd2839146bae2dcc05949b58dd38eead#diff-70887b3a407a875b71e9cb77b627856eL366

Reply:
cc @jplevyak. I know that absl flat map is faster here but we can't use it.